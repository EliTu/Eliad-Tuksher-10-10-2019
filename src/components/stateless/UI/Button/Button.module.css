@import '../../../../variables.css';

.ButtonStyles {
    width: 20%;
    height: 50px;
    background: none;
    border: none;
    border-top: 2px solid;
    border-bottom: 2px solid;
    border-right: 2px solid;
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
    border-color: var(--color-main-black);
    margin: 0 auto;
    padding: 0.2rem;
}