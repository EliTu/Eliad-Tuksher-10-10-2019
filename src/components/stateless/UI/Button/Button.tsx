import React from 'react';
import styles from './Button.module.css';

interface IProps {
	children: string;
}

const Button: React.FC<IProps> = ({ children }) => {
    const { ButtonStyles } = styles;
    
	return (
		<button className={ButtonStyles} type="button">
			{children}
		</button>
	);
};

export default Button;
