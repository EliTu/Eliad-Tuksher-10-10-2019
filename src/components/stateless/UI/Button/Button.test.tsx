import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Button from './Button';
import initialShallowRender from '../../../../utilities/test-utilities/initialShallowRender';

describe('Button component', () => {
	let component: ShallowWrapper;
	const clickFn: () => void = jest.fn();

	beforeEach(
		() =>
			(component = shallow(
				<Button handleButtonClick={clickFn}>abc</Button>
			))
	);

	it('should render without errors', () =>
		initialShallowRender(component, '.ButtonStyles'));

	it('should render children props with a string type', () => {
		expect(component.children().text()).toBe('abc');
		expect(component.children().text()).not.toBe('undefined');
	});

	it('should call the onClick callback function upon a click on the button', () => {
		component.simulate('click');

		expect(clickFn).toHaveBeenCalled();
		expect(clickFn).toHaveBeenCalledTimes(1);
	});
});
