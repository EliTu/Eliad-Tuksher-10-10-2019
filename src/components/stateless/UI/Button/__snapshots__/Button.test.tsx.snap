// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render without errors 1`] = `
<button
  className="ButtonStyles"
  type="button"
>
  abc
</button>
`;
